roles/libvirt_manager/tasks/create_networks.yml

```diff
@@ -100,3 +100,18 @@
   loop: "{{ networks }}"
   loop_control:
     label: "{{ item.name }}"
+
+- name: Extract IP address from network bridges
```

Comment:
This does not work becayse bridges are not created by libvirt - i.e the bridge is already created with some other tooling and libvirt XML only refers to the existing bridge without any knowledge if there is an IP configured on the bridge or not.

```
2024-05-13 01:48:18.180641 | controller | TASK [libvirt_manager : Extract IP address from network bridges] ***************
2024-05-13 01:48:19.228888 | controller | failed: [instance] (item=cifmw-intnet-0) => changed=false
2024-05-13 01:48:19.230235 | controller |   ansible_loop_var: item
2024-05-13 01:48:19.230302 | controller |   item:
2024-05-13 01:48:19.230320 | controller |     name: cifmw-intnet-0
2024-05-13 01:48:19.230335 | controller |     xml: |-
2024-05-13 01:48:19.230359 | controller |       <network>
2024-05-13 01:48:19.230373 | controller |         <name>cifmw-intnet-0</name>
2024-05-13 01:48:19.230389 | controller |         <bridge name='cifmw-intnet-0' stp='on' delay='0'/>
2024-05-13 01:48:19.230450 | controller |       </network>
2024-05-13 01:48:19.230469 | controller |   msg: Xpath /network/ip does not reference a node!
2024-05-13 01:48:19.494578 | controller | failed: [instance] (item=cifmw-intnet-1) => changed=false
2024-05-13 01:48:19.494623 | controller |   ansible_loop_var: item
2024-05-13 01:48:19.494981 | controller |   item:
2024-05-13 01:48:19.495086 | controller |     name: cifmw-intnet-1
2024-05-13 01:48:19.496939 | controller |     xml: |-
2024-05-13 01:48:19.496987 | controller |       <network>
2024-05-13 01:48:19.497000 | controller |         <name>cifmw-intnet-1</name>
2024-05-13 01:48:19.497013 | controller |         <bridge name='cifmw-intnet-1' stp='on' delay='0'/>
2024-05-13 01:48:19.497024 | controller |       </network>
2024-05-13 01:48:19.497035 | controller |   msg: Xpath /network/ip does not reference a node!
2024-05-13 01:48:19.499559 | controller | ok: [instance] => (item=cifmw-public)
2024-05-13 01:48:19.499825 | controller | ok: [instance] => (item=cifmw-osp_trunk)
```

Reply:
Yes, this is getting the IP defined by the user like [1], the issue is I didn't handle when the IP address isn't defined. I have added that now so those networks are skipped thank you!

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/52261a3c2e260af2379c6fb0f45120dd204c66b5/scenarios/reproducers/3-nodes.yml#L71

Reply:
Let's not change it now, but I think the dev-scripts role creates bridges with nmstate and then use those bridges in libvirt. The bridge does have IP - but libvirt XML does not show it. If we run into a case where we need to use such a bridge we could use the bridge name and do `ip -json addr show` or do a target gather facts on networks stuff to get the IP.